runtime/src/main/java/com/datastax/oss/quarkus/runtime/api/session/QuarkusCqlSession.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.runtime.api.session;
+
+import com.datastax.dse.driver.api.core.cql.continuous.ContinuousSession;
+import com.datastax.dse.driver.api.core.graph.GraphStatement;
+import com.datastax.dse.driver.api.core.graph.reactive.ReactiveGraphNode;
+import com.datastax.dse.driver.api.core.graph.reactive.ReactiveGraphResultSet;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.Statement;
+import com.datastax.oss.quarkus.runtime.api.reactive.MutinyContinuousReactiveResultSet;
+import com.datastax.oss.quarkus.runtime.api.reactive.MutinyGraphReactiveResultSet;
+import com.datastax.oss.quarkus.runtime.api.reactive.MutinyReactiveResultSet;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.smallrye.mutiny.Multi;
+
+public interface QuarkusCqlSession extends CqlSession {
```

Comment:
I think it would be better to follow existing patterns and extract all the Mutiny-related operations as default methods in a standalone interface:
```java
public interface MutinyReactiveSession extends Session {
  default MutinyReactiveResultSet executeReactive(@NonNull String query) { ... }
  ...
}

public interface QuarkusCqlSession extends CqlSession, MutinyReactiveSession;
```
The advantage is that it lets you expose just what you need: if a client application is expected to only ever use reactive methods, you can downcast to `MutinyReactiveSession` and the client code will only see the methods it really needs.

The downside is that you'll need new `RequestProcessor`s for those methods (you can't delegate to the `CqlSession` reactive methods because the new interface doesn't inherit them anymore). But those implementations are trivial (see `CqlRequestReactiveProcessor` for example). And you already have a custom context and builder, so registering them won't be too much work either.

Reply:
@adutra wdyt about it?

Reply:
I see a difference in the pattern here: we are not creating new ways of executing queries, we are simply narrowing down the return type of a few existing methods from `ReactiveResultSet` to `MutinyReactiveResultSet`. Creating new request processors for these cases seem a bit overkill imo. But I think it should be possible to combine the best of both worlds: we can make `MutinyReactiveSession` extend `ReactiveSession`. By doing so we can isolate mutiny methods in specific interfaces and still avoid the cost of creating processors for them. I took a stab at that, see commit edd1333.

Reply:
@adutra for me it looks clean. What I like about this solution is that it using the same pattern that we did use for creating `CqlSesssion` interface itself.

Reply:
Hrmm ok, I had missed the fact that we are replacing the reactive execution methods, not creating new ones.
In that case I agree @adutra's solution works, it does not introduce coupling between unrelated interfaces.